livekit_agent.proto

```diff
@@ -44,6 +44,9 @@ message WorkerMessage {
         // worker can update its status to the server, including taking itself out of the pool
         UpdateWorkerStatus status = 3;
         JobStatusUpdate job_update = 4;
+	PingRequest ping = 5;
+	SimulateJobRequest simulate_job = 6;
+	MigrateJobRequest migrate_job = 7;
```

Comment:
Do we need a MigrateJobResponse? Wondering how to minimize the time the agent is not running

Proposal of migration flow:
1. Overloaded Worker Requests a Migration
2. SFU receives migration, creates migration job request (different than normal job request because it should have the same agent identity?)
3. New worker picks up resumed job and connects
4. Overloaded worker disconnects because of duplicate identity?

Reply:
So when migrating, it comes from the current worker, so it'll just disconnect. If the worker is overloaded something is not working anyway

Reply:
The goal here is to remove a job asap to make the other running jobs run smoothly

Reply:
Hrmm maybe the migration should be thought of as more eager (to prevent downtime) and the overloaded case is considered more of a failure case captured with health checks?

Migration might also update the job cost so the SFU can schedule with more up-to-date information. Job cost might be something we want to be updateable (i.e. the egress problem of not knowing how much decoding it will have to do upfront).

Wdyt?

Reply:
I think every app should optimize it to not happen. Also the cost is defined by the local worker. So the cost is someway already changed

Reply:
I'm just trying to think of the common case for migration. Consider an agent that takes 30 second clips that composes audio/video tracks together. As new users join, the agent will use more and more load. It's not easy to predict how much load the agent will consume up-front. 

So:
- we may want to update the cost periodically so the SFU has an accurate view (we can provide a default cost updater based on CPU / #Cores)
- as a single agent takes up more and more resources, the worker may want to migrate smaller jobs off before the worker becomes overloaded

Reply:
The cost is only an abstract value. The SFU already know the load with our ping requests 🤷

Reply:
I think u can have jobs that take more resource over time, but you can plan it with the costs